### PR TITLE
Escape backslashes in case the user is authenticating with Active Direct...

### PR DESCRIPTION
**Pull request by chrisbaldauf**
...ory domain authentication (e.g. 'domain\username')

See [#38](https://github.com/tknerr/vagrant-managed-servers/issues/38) for more details.

### CODE REVIEW

File: lib/vagrant-managed-servers/action/sync_folders.rb [hunk 1]
```diff
@@ -62,6 +62,10 @@ def call(env)
                                 :hostpath => hostpath,
                                 :guestpath => guestpath))
 
+            # Escape backlashes in case the user is trying to authenticate using
+            # Active Directory domain credentials (e.g. "domain\username")
+            ssh_info.gsub!("\\", "\\\\\\\\")
```
Comment (tknerr) [inline lib/vagrant-managed-servers/action/sync_folders.rb]: that looks totally weird, but in fact I did something similar [in some other place](https://github.com/tknerr/vagrant-workflow-tests/blob/master/spec/helpers.rb#L29)... ;-)

Comment (tknerr) [inline lib/vagrant-managed-servers/action/sync_folders.rb]: Oh, and shouldn't that be `ssh_info[:username] .gsub!(..)` instead?

Comment (chrisbaldauf) [inline lib/vagrant-managed-servers/action/sync_folders.rb]: You are absolutely right. I did it once in a branch and then remembered that I needed to update my fork and opted to just redo the changes. Update coming.

**Comment by tknerr** [comment]
As you said in #38, if this happens *only* on Windows hosts, we should add a `if Vagrant::Util::Platform.windows?` guard too. 

Would be awesome if you could double-check how it behaves from your Mac, so that we don't fix it for windows but break it for other platforms. I don't have a setup here to verify this otherwise :-/

Let me know and I'll cut a new release after the merge.

**Comment by chrisbaldauf** [comment]
I'll take a look later on and let you know. Thanks!

**Comment by chrisbaldauf** [comment]
I went back to the drawing board on this one and instead of escaping backslashes, I just surrounded the username in single quotes for the `chown` and `rsync (ssh)` commands. I moved the username from `user@machine` to `-l 'user'` format because I thought it felt more obvious that we were doing something special, but I can change that back if you like.

I've tested this on Windows 7 and Mac OSX 10.10 targeting an ubuntu managed server with domain usernames (containing a backslash) and regular (no backslash) usernames and everything seemed to work fine.

**Comment by tknerr** [comment]
Thanks @chrisbaldauf. So obvious that even 4 eyes didn't see that in the first place ;-)

**Comment by chrisbaldauf** [comment]
Thanks @tknerr!

**Pull request merged**